src/qibo/hamiltonians.py

```diff
@@ -186,3 +188,129 @@ def MaxCut(nqubits, dense=True):
     if dense:
         return ham.dense
     return ham
+
+
+from qibo.symbols import X as tspX
+from qibo.symbols import Y as tspY
+from qibo.symbols import Z as tspZ
+class TSP:
+    """
+    This is a TSP class that enables us to implement TSP according to
+    https://arxiv.org/pdf/1709.03489.pdf by Hadfield (2017).
```

Comment:
Please follow the sphinx syntax for links
```suggestion
    `arxiv:1709.03489<https://arxiv.org/abs/1709.03489>`_ by Hadfield (2017).
```
so that this is nicely rendered in the docs.